src/Avalonia.Controls/DateTimePickers/LoopingSelector.cs

```diff
@@ -0,0 +1,711 @@
+﻿using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Templates;
+using Avalonia.Controls.Utils;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+
+namespace Avalonia.Controls.Primitives
+{
+
+    public delegate void SelectionChangedEventHandler(object sender, SelectionChangedEventArgs e);
+
+    /// <summary>
+    /// An items control with the ability for infinite looping
+    /// <para>
+    /// Supports UI virtualization, though doesn't inherit from ItemsControl so we manage containers and
+    /// realized items ourselves
+    /// </para>
+    /// <para>
+    /// In WinUI, this control isn't usable outside of the Date/Time Pickers, but here it technically can be
+    /// Note, it's default behavior is for the pickers though. 
+    /// </para>
+    /// </summary>
+    public class LoopingSelector : TemplatedControl
```

Comment:
I feel like this control is simultaneously too general-purpose and too specific to date/time pickers:

- It needs a `LoopingSelectorItem` to be created for each item to be displayed. This results in **201** `DatePickerPresenterItem`s being created for selecting years using the default settings! This is excessive given that only 9 year items are displayed on my screen at any one time
- It's not general purpose enough in the the sense that a looping control would be useful for a variety of situations (image gallery, rolodex, album display) but this control can only arrange things vertically in a stack

Reply:
Instead of this halfway house, what about creating a simple control tailored to date/time pickers? Something like https://gist.github.com/grokys/6f9d508241cf7837f45c76c2a1eabf42 as a quick sketch?